### PR TITLE
Support elasticsearch

### PR DESCRIPTION
**Pull request by oliverpei**
make testacc TESTARGS=" -run=TestAccTencentCloudElasticsearch"
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./... -v -run=TestAccTencentCloudElasticsearch -timeout 120m
?       github.com/terraform-providers/terraform-provider-tencentcloud  [no test files]
testing: warning: no tests to run
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/gendoc   0.015s [no tests to run]
=== RUN   TestAccTencentCloudElasticsearchInstancesDataSource
=== PAUSE TestAccTencentCloudElasticsearchInstancesDataSource
=== RUN   TestAccTencentCloudElasticsearchInstance
=== PAUSE TestAccTencentCloudElasticsearchInstance
=== CONT  TestAccTencentCloudElasticsearchInstancesDataSource
=== CONT  TestAccTencentCloudElasticsearchInstance
--- PASS: TestAccTencentCloudElasticsearchInstancesDataSource (782.54s)
--- PASS: TestAccTencentCloudElasticsearchInstance (1580.23s)
PASS
ok      github.com/terraform-providers/terraform-provider-tencentcloud/tencentcloud     1580.260s

### CODE REVIEW

File: tencentcloud/resource_tc_elasticsearch_instance.go [hunk 1]
```diff
@@ -0,0 +1,609 @@
+/*
+Provides a elasticsearch instance resource.
```
Comment (gailwang) [inline tencentcloud/resource_tc_elasticsearch_instance.go:2]: an

File: tencentcloud/resource_tc_elasticsearch_instance.go [hunk 2]
```diff
@@ -0,0 +1,609 @@
...
+			},
+			"availability_zone": {
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
+				Description: "Availability zone.",
+			},
+			"version": {
+				Type:        schema.TypeString,
+				Required:    true,
```
Comment (gailwang) [inline tencentcloud/resource_tc_elasticsearch_instance.go:75]: 是否加个校验

File: tencentcloud/resource_tc_elasticsearch_instance.go [hunk 3]
```diff
@@ -0,0 +1,609 @@
...
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      ES_NODE_TYPE_HOT_DATA,
+							ValidateFunc: validateAllowedStringValue(ES_NODE_TYPE),
+							Description:  "Node type. Valid values are `hotData`, `warmData` and `dedicatedMaster`, and default value is 'hotData`.",
+						},
+						"disk_type": {
+							Type:         schema.TypeString,
+							Optional:     true,
+							Default:      CVM_DISK_TYPE_CLOUD_SSD,
```
Comment (gailwang) [inline tencentcloud/resource_tc_elasticsearch_instance.go:182]: 这里CVM是不是笔误

File: tencentcloud/resource_tc_elasticsearch_instance.go [hunk 4]
```diff
@@ -0,0 +1,609 @@
...
+				for _, item := range infos {
+					value := item.(map[string]interface{})
+					info := es.ZoneDetail{
+						Zone:     helper.String(value["availability_zone"].(string)),
+						SubnetId: helper.String(value["subnet_id"].(string)),
+					}
+					request.MultiZoneInfo = append(request.MultiZoneInfo, &info)
+				}
+			} else {
+				return fmt.Errorf("elasticsearch multi_zone_infos can not be enpty when deploy mode is %d", deployMode)
```
Comment (gailwang) [inline tencentcloud/resource_tc_elasticsearch_instance.go:285]: empty

File: tencentcloud/resource_tc_elasticsearch_instance.go [hunk 5]
```diff
@@ -0,0 +1,609 @@
...
+		info := make(map[string]interface{}, 2)
+		info["availability_zone"] = item.Zone
+		info["subnet_id"] = item.SubnetId
+		multiZoneInfos = append(multiZoneInfos, info)
+	}
+	_ = d.Set("multi_zone_infos", multiZoneInfos)
+
+	nodeInfoList := make([]map[string]interface{}, 0, len(instance.NodeInfoList))
+	for _, item := range instance.NodeInfoList {
+		info := make(map[string]interface{}, 4)
```
Comment (gailwang) [inline tencentcloud/resource_tc_elasticsearch_instance.go:418]: 4or5

File: tencentcloud/resource_tc_elasticsearch_instance.go (hunk 2 above)
Comment (oliverpei) [reply tencentcloud/resource_tc_elasticsearch_instance.go:75]: 这里不用加校验，要不后面API支持更多版本了需要一直更新验证条件

File: tencentcloud/resource_tc_elasticsearch_instance.go (hunk 3 above)
Comment (oliverpei) [reply tencentcloud/resource_tc_elasticsearch_instance.go:182]: 这里用的就是CVM硬盘类型的常量

File: tencentcloud/provider.go [hunk 6]
```diff
@@ -201,7 +201,14 @@ Direct Connect Gateway(DCG)
 
   Resource
     tencentcloud_dc_gateway
-    tencentcloud_dc_gateway_ccn_route
+	tencentcloud_dc_gateway_ccn_route
```
Comment (likexian) [inline tencentcloud/provider.go]: 缩进不对？

**Pull request merged**